src/qat/purr/compiler/devices.py

```diff
@@ -404,6 +405,29 @@ def __init__(
                 f"channel with id {self.full_id()}."
             )
 
+    @property
+    def id(self):
+        return self._id
+
+    @id.setter
+    def id(self, val):
+        self._id = val
+        if hasattr(self, "_physical_channel"):
```

Comment:
is this for the PulseChannelView problem ?

Reply:
Thanks for checking it over! :smile: 

It's because constructing the full id needs both `_id` and `_physical_channel` to be set: in the constructor, `id` is called before the physical channel is set.